lib/spaceship/tunes/build_train.rb

```diff
@@ -83,17 +83,25 @@ def update_testing_status!(new_value, testing_type, build = nil)
 
         build ||= latest_build if testing_type == 'external'
 
-        data['trains'].each do |train|
-          train["#{testing_type}Testing"]['value'] = false
-          train["#{testing_type}Testing"]['value'] = new_value if train['versionString'] == version_string
-
-          # also update the builds
-          train['builds'].each do |b|
-            next if b["#{testing_type}Testing"].nil?
-            next if build.nil?
-            next if b["buildVersion"] != build.build_version
-            b["#{testing_type}Testing"]['value'] = false
-            b["#{testing_type}Testing"]['value'] = new_value if b['trainVersion'] == version_string
+        # Delete the irrelevant trains and update the relevant one to enable testing
+        data['trains'].delete_if do |train|
+          if train['versionString'] != version_string
+            true
+          else
+            train["#{testing_type}Testing"]['value'] = false
+            train["#{testing_type}Testing"]['value'] = new_value if train['versionString'] == version_string
```

Comment:
if you delete, you probably do not need to check for versionString anymore (here and below)